### DISCUSSION TITLE
Debug nodejs application

### DISCUSSION THREAD

**Discussion by SofiaTa** [eclipse-theia/theia#8650] [Q&A]
Hi,

i've deployed very simple nodejs/express app to k8s which is working. im able to see the output in the browser.

I've exposed port `9229` in the docker file and my app start with 
`"start": "node --inspect index.js"`

Im using the k8s plugin and from workloads->pods->choose the required pod and use debug(attach)
choose the pod -> nodejs and got the following logs:

```
witching node to debug mode in container
Trying to detect remote root.
Got remote root from container: /opt/node_app/app
[Set up port forwarding 2020-10-19 15:11:19.232]
Setting up port forwarding on pod myapp-host-7c54cb7-jjr7q...
Created port-forward **10000:9229** 
Waiting for port forwarding to be ready...
Port forwarding is ready.
[Start debug session 2020-10-19 15:11:19.766]
Starting debug session...
**The debug session has started. Your application is ready for you to debug**

``` 

in addition, I got a pop-up with the message:

`A service is available on port 10000`

I've created the following launch.json

```
  {"version": "0.2.0",
  "configurations": [
    {
      "type": "node",
      "request": "attach",
      "name": "Debug",
      "port": 9229,

      "openDebug": "openOnDebugBreak",
      "internalConsoleOptions": "openOnFirstSessionStart",
      "localRoot": "${workspaceRoot}/myapp",
      "remoteRoot": "/app"
    }
  ]
}

```

when click on the debug button I got error:
`Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:9229).` am I missing something?  is it a bug?

**Comment by SofiaTa** [comment]
Could I get some help? is it a bug? or am I using the configuration in a wrong way as from VSCode I was able to debug nodejs app inside k8s

**Reply by vince-fugnitto** [reply]
@SofiaTa I'd love to be able to help, unfortunately I do not have knowledge of k8s nor can I easily reproduce the issue.
If you believe it is a bug please open a relevant [issue](https://github.com/eclipse-theia/theia/issues/new/choose) with as much detail as possible.

@eclipse-theia/core does anyone have kubernetes knowledge that may be able to help?
